### ISSUE TITLE
Burst compiler bug when converting constant uint to ulong?

### ISSUE THREAD

**Issue by xfxyjwf** [Unity-Technologies/EntityComponentSystemSamples#76]
The following assignment doesn't work correctly when burst compile is enabled:
```C#
ulong value = uint.MaxValue;
```
The correct value should be 4294967295, but when burst compile is enabled it will produce 18446744073709551615.

A complete example:
```C#
    public struct Data : IComponentData
    {
        public ulong Value;
    }
    public class TestBurstComponentSystem : JobComponentSystem
    {
        [BurstCompile]
        private struct Job : IJobProcessComponentData<Data>
        {
            public void Execute(ref Data data)
            {
                data.Value = uint.MaxValue;
            }
        }
 
        protected override JobHandle OnUpdate(JobHandle inputDeps)
        {
            return new Job().Schedule(this, inputDeps);
        }
    }
```
**By enabling/disabling burst jobs, you will get different values out of the Data component.**

I have created a unity project that can reproduce this incorrect behavior here: https://github.com/xfxyjwf/burstbug

More detailed report I posted on the unity forum: https://forum.unity.com/threads/bug-burst-compiler-bug-when-converting-uint-constants-to-ulong.584173/

Unity Version: the issue can be reproduced with both 2018.2.10f1 and 2018.2.16f1
Entities: 0.0.12-preview.20
Burst: 0.2.4-preview.37
IncrementalCompiler: 0.0.42-preview.24

**Comment by xoofx** [comment]
Our apologize for this issue, this should be fixed in an upcoming `0.2.4-preview.38` available later today

**Comment by xfxyjwf** [comment]
Confirmed the issue is fixed in 0.2.4-preview.38
